File: src/simulation/culture/group/cultureaspect/Ritual.java
package simulation.culture.group.cultureaspect;

import simulation.culture.group.Group;
import simulation.culture.group.reason.Reason;

public abstract class Ritual extends AbstractCultureAspect {
    Reason reason;

    public Ritual(Group group, Reason reason) {
        super(group);
        this.reason = reason;
    }

    public Reason getReason() {
        return reason;
    }
}

File: src/io/tashtabash/simulation/culture/group/cultureaspect/Ritual.kt
package io.tashtabash.simulation.culture.group.cultureaspect

import io.tashtabash.simulation.culture.group.centers.Group
import io.tashtabash.simulation.culture.group.reason.Reason


abstract class Ritual(var reason: Reason) : CultureAspect {
    abstract override fun adopt(group: Group): Ritual?
}
